File: app/src/integration/java/com/mycompany/myapp/app/ApplicationComponent.java
package com.mycompany.myapp.app;

import android.app.Application;

import com.mycompany.myapp.data.DataModule;
import com.mycompany.myapp.data.api.github.GitHubBusService;
import com.mycompany.myapp.data.api.github.GitHubModule;
import com.mycompany.myapp.data.api.github.MockGitHubModule;
import com.mycompany.myapp.modules.CrashReporterModule;
import com.mycompany.myapp.monitoring.CrashReporter;
import com.mycompany.myapp.monitoring.LoggerModule;
import com.squareup.otto.Bus;

import javax.inject.Singleton;

import dagger.Component;
import retrofit.client.Client;
import timber.log.Timber.Tree;

@Singleton
@Component(modules = {
        AndroidModule.class,
        LoggerModule.class,
        CrashReporterModule.class,
        BusModule.class,
        MockGitHubModule.class
})
public interface ApplicationComponent {
    Application application();

    Tree tree();

    CrashReporter crashReporter();

    Bus bus();

    GitHubBusService gitHubBusService();

    void inject(MainApplication application);
}

File: app/src/main/java/com/atomicrobot/carbon/app/ApplicationComponent.kt
package com.atomicrobot.carbon.app

import com.atomicrobot.carbon.data.DataModule
import com.atomicrobot.carbon.modules.CrashReporterModule
import com.atomicrobot.carbon.monitoring.LoggerModule
import com.atomicrobot.carbon.ui.ViewModelFactoryModule
import com.atomicrobot.carbon.StartActivity
// GENERATOR - MORE IMPORTS //
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
        VariantModule::class,
        AndroidModule::class,
        AppModule::class,
        LoggerModule::class,
        CrashReporterModule::class,
        DataModule::class,
        ViewModelFactoryModule::class])
interface ApplicationComponent : VariantApplicationComponent {
    fun inject(application: MainApplication)

    fun inject(activity: StartActivity)
    // GENERATOR - MORE ACTIVITIES //
}